src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -926,3 +928,198 @@ def acquireTiledArea(streams, stage, area, overlap=0.2, settings_obs=None, log_p
+        self._weaver = weaver
+
+    def cancel(self, future):
+        """
+        Canceler of acquisition task.
+        """
+        logging.debug("canceling acquisition overview...")
+
+        with future._task_lock:
+            if future._task_state == FINISHED:
+                return False
+            future._task_state = CANCELLED
+            future.running_subf.cancel()
+            logging.debug("acquisition overview cancelled.")
+        return True
+
+    def estimate_time(self, roi_idx=0, actual_time_per_roi=None) -> float:
+        """
+        Estimates the time for the rest of the acquisition.
+
+        :param roi_idx: (int) index of the current roi
+        :param actual_time_per_roi: (float) the actual time spent for the current roi
+        :return: (float) the estimated time for the rest of the acquisition
+        """
+        remaining_rois = (len(self.areas) - roi_idx)
+
+        if actual_time_per_roi:
+            acquisition_time = actual_time_per_roi * remaining_rois
+        else:
+            acquisition_time = estimate_autofocus_in_roi_time(self.n_focus_points, self._ccd) * remaining_rois
+
+        return acquisition_time
+
+    def run(self):
+        """
+        The main function of the task class, which will be called by the future asynchronously
+        """
+        self._future._task_state = RUNNING
+        da_rois = []
+        try:
+            actual_time_per_roi = None
+            start_time = time.time()
+            # create a for loop for roi to create sub futures
+            for idx, roi in enumerate(self.areas):
+                remaining_t = self.estimate_time(idx, actual_time_per_roi)
+                self._future.set_end_time(time.time() + remaining_t)
+
+                # cancel the sub future
+                with self._future._task_lock:
+                    if self._future._task_state == CANCELLED:
+                        raise CancelledError()
+
+                # run autofocus for the selected roi
+                self._future.running_subf = autofocus_in_roi(roi,
+                                                             self._stage,
+                                                             self._ccd,
+                                                             self._focus,
+                                                             self.focus_rng,
+                                                             self.n_focus_points,
+                                                             self.conf_level)
+
+                logging.debug(f"Autofocus is running for roi number {idx} with {roi} values")
+
+                try:
+                    # TODO use the "estimated time * 3 + 1" as timeout error
+                    focus_points = self._future.running_subf.result()
+                except Exception:
+                    logging.debug(
+                        f"Autofocus within roi failed for roi number {idx} with {roi} values")
```

Comment:
same:
`roi` is a bounding box of the overview-area containing the left, top, right, bottom positions of this area. I think this debug message can be a bit more clear:
```suggestion
                logging.debug(f"Autofocus is running for roi number {idx}, with bounding box: {roi} [m]")
```